packages/services/api/src/shared/aws.ts

```diff
@@ -0,0 +1,458 @@
+    datetime?: string;
+    signQuery?: boolean;
+    appendSessionToken?: boolean;
+    allHeaders?: boolean;
+    singleEncode?: boolean;
+  };
+};
+
+export class AwsClient {
+  private secretAccessKey: string;
+  private accessKeyId: string;
+  private sessionToken?: string;
+  private service?: string;
+  private region?: string;
+  private cache: Map<string, ArrayBuffer>;
+  private retries: number;
+  private initRetryMs: number;
+
+  constructor({
+    accessKeyId,
+    secretAccessKey,
+    sessionToken,
+    service,
+    region,
+    cache,
+    retries,
+    initRetryMs,
+  }: {
+    accessKeyId: string;
+    secretAccessKey: string;
+    sessionToken?: string;
+    service?: string;
+    region?: string;
+    cache?: Map<string, ArrayBuffer>;
+    retries?: number;
+    initRetryMs?: number;
+  }) {
+    if (accessKeyId == null) throw new TypeError('accessKeyId is a required option');
+    if (secretAccessKey == null) throw new TypeError('secretAccessKey is a required option');
+    this.accessKeyId = accessKeyId;
+    this.secretAccessKey = secretAccessKey;
+    this.sessionToken = sessionToken;
+    this.service = service;
+    this.region = region;
+    this.cache = cache || new Map();
+    this.retries = retries != null ? retries : 10; // Up to 25.6 secs
+    this.initRetryMs = initRetryMs || 50;
+  }
+
+  async sign(input: RequestInfo, init?: AwsRequestInit) {
+    if (input instanceof Request) {
+      const { method, url, headers, body } = input;
+      init = Object.assign({ method, url, headers }, init);
+      if (init.body == null && headers.has('Content-Type')) {
+        init.body =
+          body != null && headers.has('X-Amz-Content-Sha256')
+            ? body
+            : await input.clone().arrayBuffer();
+      }
+      input = url;
+    }
+    const signer = new AwsV4Signer(Object.assign({ url: input }, init, this, init && init.aws));
+    const signed = Object.assign({}, init, await signer.sign());
+    delete signed.aws;
+    try {
+      return new Request(signed.url.toString(), signed);
+    } catch (e) {
+      if (e instanceof TypeError) {
+        // https://bugs.chromium.org/p/chromium/issues/detail?id=1360943
+        return new Request(signed.url.toString(), Object.assign({ duplex: 'half' }, signed));
+      }
+      throw e;
+    }
+  }
+
+  async fetch(input: RequestInfo, init: AwsRequestInit): Promise<Response> {
+    for (let i = 0; i <= this.retries; i++) {
+      const fetched = fetch(await this.sign(input, init));
+      if (i === this.retries) {
+        return fetched; // No need to await if we're returning anyway
+      }
+      const res = await fetched;
+      if (res.status < 500 && res.status !== 429) {
+        return res;
+      }
+      await new Promise(resolve =>
+        setTimeout(resolve, Math.random() * this.initRetryMs * Math.pow(2, i)),
+      );
+    }
+    throw new Error('An unknown error occurred, ensure retries is not negative');
+  }
+}
```

Comment:
@kamilkisiela This thing has 5XX reties included

Reply:
500 retries is a lot (joke)

Maybe we could use https://github.com/elbywan/wretch or https://github.com/developit/redaxios to not do this the manual way? Not saying we should, just wanted to share the libs with you, you decide.

Reply:
@kamilkisiela I am a fan of being explicit over abstractions so I won't wadd this myself

Reply:
why Yoga then? :))